llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -14315,6 +14316,18 @@ SDValue DAGCombiner::reduceLoadWidth(SDNode *N) {
         if ((ExtVT.getScalarSizeInBits() > MaskedVT.getScalarSizeInBits()) &&
             TLI.isLoadExtLegal(ExtType, SRL.getValueType(), MaskedVT))
           ExtVT = MaskedVT;
+      } else if (ExtType == ISD::ZEXTLOAD &&
+                 ShiftMask.isShiftedMask(Offset, ActiveBits) &&
+                 (Offset + ShAmt) < VT.getSizeInBits()) {
+        EVT MaskedVT = EVT::getIntegerVT(*DAG.getContext(), ActiveBits);
+        // If the mask is shifted we can use a narrower load and a shl to insert
+        // the trailing zeros.
+        if (((Offset + ActiveBits) <= ExtVT.getSizeInBits()) &&
```

Comment:
ExtVT.getScalarSizeInBits() - not sure if anything other than the isa<ConstantSDNode> ensure this is for scalar only?

Reply:
It is actually guarded on line 14189 (there is an early out for vector loads).

Reply:
I think I meant to use getScalarSizeInBits(). I'll change it.

Reply:
`ExtVT.getSizeInBits()` -> `ExtVT.getScalarSizeInBits()` (note this is already used above)